docs/patterns/sidecar_injection.md

```diff
@@ -4,6 +4,54 @@ Services participating in the service mesh communicate via sidecar proxies insta
 ## Automatic Sidecar Injection
 Automatic sidecar injection is currently the only way to inject sidecars into the service mesh. Sidecars can be automatically injected into applicable Kubernetes pods using a mutating webhook admission controller provided by OSM.
 
-Each OSM instance is given a unique ID on installation. This ID is used while labeling namespaces as a way to configure OSM to monitor the namespaces. When a namespace is labeled with `openservicemesh.io/monitored-by=<mesh-name>`, pods deployed in the monitored namespaces are automatically injected with sidecars by the corresponding OSM instance.
+Automatic sidecar injection can be configured per namespace as a part of enrolling a namespace into the mesh, or later using the Kubernetes API. Automatic sidecar injection can be enabled either on a per namespace or per pod basis by annotating the namespace or pod resource with the sidecar injection annotation. Individual pods and namespaces can be explicitly configured to either enable or disable automatic sidecar injection, giving users the flexibility to control sidecar injection on pods and namespaces.
 
-Since sidecars are automatically injected to pods deployed in OSM monitored namespaces, pods that should not be a part of the service mesh but belong to monitored namespaces need to be explicitly annotated to disable automatic sidecar injection. Using the annotation `"openservicemesh.io/sidecar-injection": "disabled"` on the POD will inform OSM to not inject the sidecar on the POD.
\ No newline at end of file
+### Enabling Automatic Sidecar Injection
+
+Automatic Sidecar injection can be enabled in the following ways:
+
+- While enrolling a namespace into the mesh using `osm` cli: `osm namespace add <namespace> --enable-sidecar-injection`
+
+- Using `kubectl` to annotate individual namespaces and pods to enable sidecar injection:
+
+  ```console
+  # Enable sidecar injection on a namespace
+  $ kubectl annotate namespace <namespace> openservicemesh.io/sidecar-injection=enabled
+  ```
+
+  ```console
+  # Enable sidecar injection on a pod
+  $ kubectl annotate pod <pod> openservicemesh.io/sidecar-injection=enabled
+  ```
+
+- Setting the sidecar injection annotation to `enabled` in the Kubernetes resource spec for a namespace or pod:
+  ```yaml
+  metadata:
+    name: test
+    annotations:
+      'openservicemesh.io/sidecar-injection': 'enabled'
+  ```
+
+  Pods will be injected with a sidecar only if the following conditions are met:
+  1. The namespace to which the pod belongs is a monitored namespace.
+  2. The pod is explicitly enabled for the sidecar injection, OR the namespace to which the pod belongs is enabled for the sidecar injection and the pod is not explicitly disabled for sidecar injection.
+
```

Comment:
What is the behavior if this monitor label doesn't exist but the sidecar injection annotation does exist on a namespace?

Reply:
The monitor label is a prerequisite for OSM to monitor any resources in a namespace, including sidecar injection.